### PR TITLE
Enable Firefox DevTools for testharness tests with --debug-test

### PR DESCRIPTION
**Pull request by jgraham**
If wptrunner is launched with --debug-test, start the DevTools when
the test window is opened.

### CODE REVIEW

File: tools/wptrunner/wptrunner/executors/executormarionette.py [hunk 1]
```diff
@@ -625,6 +626,30 @@ def pdf_to_png(self, pdf_base64, page_ranges):
...
             _switch_to_window(self.marionette, handle)
 
+
+class MarionetteDebugProtocolPart(DebugProtocolPart):
+    def setup(self):
+        self.marionette = self.parent.marionette
+
+    def load_devtools(self):
+        with self.marionette.using_context(self.marionette.CONTEXT_CHROME):
+                self.parent.base.execute_script("""
```
Comment (stephenmcgruer) [inline tools/wptrunner/wptrunner/executors/executormarionette.py:636]: Nit; overindented.

**Review by stephenmcgruer — APPROVED** [review]
Mostly rubber-stamp LGTM, as I am not familiar with Firefox's devtools JS.

**Comment by jgraham** [comment]
Thanks for the review :) I've asked @juliandescottes to cross check the js to launch devtools.

**Review by juliandescottes — APPROVED** [review]
Thanks! Looks fine to me. Just one question about using the "bottom" host, but I don't really know in which scenario this will be used.

In the long run having this JS here can be fragile, I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1688688 so that we can provide a single method that does that instead, Will be easier to maintain :)

File: tools/wptrunner/wptrunner/executors/executormarionette.py [hunk 2]
```diff
@@ -625,6 +626,30 @@ def pdf_to_png(self, pdf_base64, page_ranges):
...
+            self.parent.base.execute_script("""
+const { require } = ChromeUtils.import("resource://devtools/shared/Loader.jsm");
+const { TargetFactory } = require("devtools/client/framework/target");
+const { gDevTools } = require("devtools/client/framework/devtools");
+
+const callback = arguments[arguments.length - 1];
+
+async function loadDevTools() {
+    const target = await TargetFactory.forTab(window.gBrowser.selectedTab);
+    await gDevTools.showToolbox(target, "webconsole", "bottom");
```
[the remark this answers is not in the document]
Comment (jgraham) [reply tools/wptrunner/wptrunner/executors/executormarionette.py:645]: testharness tests usually aren't *that* sensitive to the page size, but nevertheless that seems like a good idea.

**Pull request merged**